src/main/java/net/minecraftforge/client/event/RegisterKeyMappingsEvent.java

```diff
@@ -5,32 +5,37 @@
 
 package net.minecraftforge.client.event;
 
-import net.minecraft.resources.ResourceLocation;
-import net.minecraftforge.client.model.IModelLoader;
-import net.minecraftforge.client.model.ModelLoaderRegistry;
+import net.minecraft.client.KeyMapping;
+import net.minecraft.client.Options;
 import net.minecraftforge.eventbus.api.Cancelable;
 import net.minecraftforge.eventbus.api.Event;
 import net.minecraftforge.fml.LogicalSide;
 import net.minecraftforge.fml.event.IModBusEvent;
 import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import org.apache.commons.lang3.ArrayUtils;
 
 /**
- * Fired when the {@link ModelLoaderRegistry} is ready for model loader registrations.
+ * Allows users to register custom {@link net.minecraft.client.KeyMapping key mappings}.
  *
- * <p>This event is not {@linkplain Cancelable cancellable}, and does not {@linkplain HasResult have a result}. </p>
+ * <p>This event is not {@linkplain Cancelable cancellable}, and does not {@linkplain HasResult have a result}.
  *
  * <p>This event is fired on the {@linkplain FMLJavaModLoadingContext#getModEventBus() mod-specific event bus},
  * only on the {@linkplain LogicalSide#CLIENT logical client}. </p>
- *
- * @see ModelLoaderRegistry#registerLoader(ResourceLocation, IModelLoader)
  */
-public class ModelRegistryEvent extends Event implements IModBusEvent
+public class RegisterKeyMappingsEvent extends Event implements IModBusEvent
 {
+    private final Options options;
+
+    public RegisterKeyMappingsEvent(Options options)
+    {
+        this.options = options;
+    }
+
     /**
-     * @hidden
-     * @see ModelLoaderRegistry#onModelLoadingStart()
+     * Registers a new key mapping.
      */
-    public ModelRegistryEvent()
+    public synchronized void register(KeyMapping key)
```

Comment:
Good point. Seems I forgot to ask about parallel dispatch. The new registration events should likely fire in parallel, since they're not inter-dependent and are already being fired with concurrent collections.

Reply:
Fixed in [7bdcf52](https://github.com/MinecraftForge/MinecraftForge/pull/8786/commits/7bdcf52de92f0c8289bf5d84a83a4063ba5213e8)